android-components/components/lib/crash/proguard-rules.pro

```diff
@@ -19,3 +19,40 @@
 # If you keep the line number information, uncomment this to
 # hide the original source file name.
 #-renamesourcefileattribute SourceFile
+
+# From https://github.com/Kotlin/kotlinx.serialization/blob/6333daf8bb4ee7e7c8cd66619c7c66cc8c69dddf/rules/common.pro
```

Comment:
> The kotlinx.serialization proguard rules aren't transitively provided, so they are copied to the proguard rules of the lib-crash library.

@afranchuk ah, it looks like the proguard rules have only recently been bundled with the library in version 1.5.0, and we're still on [1.4.1](https://github.com/mozilla-mobile/firefox-android/blob/main/android-components/plugins/dependencies/src/main/java/DependenciesPlugin.kt#L19). See release notes for [1.5.0](https://github.com/Kotlin/kotlinx.serialization/releases) (need to scroll down to the first RC which mentions it).

Let's try and upgrade to 1.5.0 instead? Then we don't need to ship our own rules. Since this dependency was only recently introduced for lib-crash anyway, there's not really a reason to wait.

Wdyt? If there's a reason not to, happy to approve and land otherwise.

Reply:
Copying relevant section from 1.5.0 release notes:

**Bundled Proguard rules**

The kotlinx-serialization-core-jvm JAR file now includes consumer Proguard rules,
so manual Proguard configuration is no longer necessary for most of the setups.

🤞

Reply:
Oh! That wasn't clear to me at all, I wish the docs mentioned that. I will try that.

Reply:
Updating to 1.5.0 works!